File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/commands/levelledmobs/subcommands/AboutSubcommand.java
package me.lokka30.levelledmobs.bukkit.commands.levelledmobs.subcommands;

import java.util.Collections;
import java.util.List;
import me.lokka30.levelledmobs.bukkit.LevelledMobs;
import me.lokka30.levelledmobs.bukkit.commands.CommandWrapper;
import me.lokka30.levelledmobs.bukkit.configs.translations.Message;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

/*
[command structure]
    index.. 0   1
    size... 1   2
            :   :
          - /lm |
          - /lm about
 */
public final class AboutSubcommand extends CommandWrapper {

    public AboutSubcommand() {
        super("about", "info", "information", "version", "ver");
    }

    @Override
    public void run(@NotNull CommandSender sender, @NotNull String[] args) {
        if(!hasPerm(sender, "levelledmobs.command.levelledmobs.about", true))
            return;

        if(args.length != 2) {
            Message.COMMAND_LEVELLEDMOBS_SUBCOMMAND_ABOUT_INVALID_USAGE.sendTo(sender);
            return;
        }

        final var pdf = LevelledMobs.getInstance().getDescription();

        Message.COMMAND_LEVELLEDMOBS_SUBCOMMAND_ABOUT_SUCCESS.sendTo(sender,
            "%version%", pdf.getVersion(),
            "%maintainers%", Message.joinDelimited(pdf.getAuthors())
        );
    }

    @Override
    public @NotNull List<String> suggest(
        @NotNull CommandSender sender,
        @NotNull String[] args
    ) {
        return Collections.emptyList();
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/me/lokka30/levelledmobs/bukkit/commands/levelledmobs/subcommands/AboutSubcommand.kt
package me.lokka30.levelledmobs.bukkit.commands.levelledmobs.subcommands

import me.lokka30.levelledmobs.bukkit.LevelledMobs
import me.lokka30.levelledmobs.bukkit.LevelledMobs.Companion.prefixSev
import me.lokka30.levelledmobs.bukkit.commands.CommandWrapper
import org.bukkit.ChatColor.AQUA
import org.bukkit.ChatColor.BOLD
import org.bukkit.ChatColor.DARK_GRAY
import org.bukkit.ChatColor.GRAY
import org.bukkit.command.CommandSender

/*
[command structure]
    index.. 0   1
    size... 1   2
            :   :
          - /lm |
          - /lm about
 */
class AboutSubcommand : CommandWrapper() {

    companion object {
        const val usage = "/lm about"

        val labels = mutableSetOf(
            "about",
            "info",
            "information"
        )
    }

    override fun labels(): MutableSet<String> {
        return labels
    }

    override fun usage(): String {
        return usage
    }

    override fun run(
        sender: CommandSender,
        args: Array<String>
    ) {
        if (!hasPerm(sender, "levelledmobs.command.levelledmobs.about", true)) return

        if (args.size != 2) {
            sender.sendMessage("${prefixSev}Invalid usage; try '${AQUA}${usage}${GRAY}'.")
            return
        }

        val pdf = LevelledMobs.instance!!.description
        sender.sendMessage(
            """
            ${AQUA}${BOLD}LevelledMobs${AQUA} v${pdf.version}
            $DARK_GRAY • ${GRAY}Spigot: ${DARK_GRAY}https://spigotmc.org/resources/74304
            $DARK_GRAY • ${GRAY}Wiki: ${DARK_GRAY}https://github.com/lokka30/LevelledMobs/Wiki
            $DARK_GRAY • ${GRAY}Source: ${DARK_GRAY}https://github.com/lokka30/LevelledMobs
            $DARK_GRAY • ${GRAY}Credits: ${DARK_GRAY}https://github.com/lokka30/LevelledMobs/wiki/Credits
            """.trimIndent()
        )
    }

}